pkg/reconciler/v1alpha1/taskrun/resources/input_resources.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2018 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	v1alpha1 "github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1"
+	listers "github.com/knative/build-pipeline/pkg/client/listers/pipeline/v1alpha1"
+	buildv1alpha1 "github.com/knative/build/pkg/apis/build/v1alpha1"
+	"go.uber.org/zap"
+)
+
+// AddInputResource will update the input build with the input resource from the task
+func AddInputResource(
+	build *buildv1alpha1.Build,
+	task *v1alpha1.Task,
+	taskRun *v1alpha1.TaskRun,
+	pipelineResourceLister listers.PipelineResourceLister,
+	logger *zap.SugaredLogger,
+) (*buildv1alpha1.Build, error) {
+
+	var gitResource *v1alpha1.GitResource
+
+	for _, input := range task.Spec.Inputs.Sources {
+		resource, err := pipelineResourceLister.PipelineResources(task.Namespace).Get(input.Name)
+		if err != nil {
+
+			logger.Errorf("Failed to reconcile TaskRun: %q failed to Get Pipeline Resource: %q", task.Name, input.Name)
+			return nil, err
+		}
+		if resource.Spec.Type == v1alpha1.PipelineResourceTypeGit {
+			gitResource = v1alpha1.NewGitResource(resource)
+			for _, trInput := range taskRun.Spec.Inputs.Resources {
+				if trInput.ResourceRef.Name == input.Name {
+					gitResource.Revision = trInput.Version
+					break
+				}
+			}
+			break
+		}
+	}
+	// default revision to master is nothing is provided
+	if gitResource.Revision == "" {
```

Comment:
hmm, i feel the defaults should be set in the GitResource Constructor. The GitResource object decalres the defaults for its fields, so the onus should be on the Constructor rather than on this method.

Reply:
the revision comes from the taskRun which happens after the constructor is called

Reply:
which is fine. We default the revision to master if resource does not `revision` defined in NewGitResource().
Then in the loop, we override it `trInput.Version`.

Reply:
ok, makes sense

Reply:
but please note that the TaskRun validation should ensure it has a version, this is a required field